### PR TITLE
Fix hardpoints not being cleared when overwriting a ship's outfits.

### PR DESCRIPTION
**Pull request by quyykk**
**Bugfix** 

When overwriting an existing ship by only changing its outfits, the outfits in the `armament` do not get cleared. This can result in ships having the wrong outfits in their armament (and if you look at the info screen you can see the wrong weapons). This results in a ship that cannot fire. If you reload your save this issue is recognized by the game with a warning and fixed (and there'll be a stray outfit entry in your outfit list).

This PR makes it so that every outfit is uninstalled from every hardpoint when only a `"outfit"`/`"gun"`/`"turret"` node is present.

sparrow

## Fix Details

There is another possible implementation for a bugfix:  In `FinishLoading` remove any hardpoint outfits that aren't in the ship's outfit list. In contrast to this PR if the outfit list changes but the weapons don't, the original hardpoint assignments will be preserved. I don't know whether this is desirable or not.

## To Reproduce

I made a plugin for this. Unzip this in your plugins folder and create a new pilot and buy the "Wanderer Sparrow", an invalid sparrow with 2 Beam Lasers in its outfit list and 2 Sunbeams in its armament.
[broken.zip](https://github.com/endless-sky/endless-sky/files/7085166/broken.zip)

### CODE REVIEW

**Comment by Amazinite** [comment]
Ref: #5649

**Comment by quyykk** [comment]
Thanks! The linked issue is now fixed.

We only actually need to clear `equipped` when we actually modify outfits or hardpoints. If not, then there is no reason at all to clear it.

**Comment by tehhowch** [comment]
>  if the outfit list changes but the weapons don't, the original hardpoint assignments will be preserved. I don't know whether this is desirable or not.

It is indeed desirable for the hardpoint assignments to be preserved.

**Comment by quyykk** [comment]
Is there anything left do to? I would really like to see this bug fixed.

**Comment by tehhowch** [comment]
I need to pull down and verify the fixes. Until then, mind merging the base / rebasing onto latest, to get an updated CI run?

File: source/Ship.cpp [hunk 1]
```diff
@@ -404,6 +404,17 @@ void Ship::Load(const DataNode &node)
 				else
 					grand.PrintTrace("Skipping invalid outfit count:");
 			}
+			
+			// Verify we have at least as many installed outfits as were identified as "equipped."
+			// If not (e.g. a variant definition), ensure FinishLoading equips into a blank slate.
+			if(!hasArmament)
+				for(const auto &it : equipped)
+					if(static_cast<int>(outfits.count(it.first)) < it.second)
```
Comment (quyykk) [inline source/Ship.cpp:412]: .... I'm dumb. @tehhowch this code has an error that I didn't notice (as usual). Sometimes I'm blind when it comes to errors in my code. 

`outfits` is a map so `count` returns either `1` or `0.... It should `find` and then use the value of the pair to compare.

Comment (tehhowch) [reply source/Ship.cpp:412]: Yeah, I missed it too when checking the bugs that were fixed by the new logic. I had even taken notice of the need for the static cast due to the type difference, but missed that it was due to comparing the wrong things. Knowing what you were intending to do, plus the wonderful language semantics (_Of course `outfits.count` counts the outfits, that's the point of the condition comparison..._), was just enough to get us both. 😅

**Pull request merged**